### ISSUE TITLE
ant test fails in dev mode

### ISSUE THREAD

**Issue by dankurka** [gwtproject/gwt#8921]
Originally reported on Google Code with ID 8967
```
Prerequisite: Creating a project with webappcreator and importing it into eclipse.

When launching test dev mode we see the following exception:

Rebinding com.google.gwt.user.client.impl.WindowImpl
   Checking rule <replace-with class='com.google.gwt.user.client.impl.WindowImplIE'/>
      [ERROR] Error while executing the JavaScript provider for property 'user.agent'
com.google.gwt.core.client.JavaScriptException: (null) __gwt_getProperty([string: 'user.agent']):
null
	at com.google.gwt.dev.shell.BrowserChannelServer.invokeJavascript(BrowserChannelServer.java:252)
	at com.google.gwt.dev.shell.ModuleSpaceOOPHM.doInvoke(ModuleSpaceOOPHM.java:137)
	at com.google.gwt.dev.shell.ModuleSpace.invokeNative(ModuleSpace.java:589)
	at com.google.gwt.dev.shell.ModuleSpace.invokeNativeObject(ModuleSpace.java:293)
	at com.google.gwt.dev.shell.ModuleSpacePropertyOracle.computePropertyValue(ModuleSpacePropertyOracle.java:152)
	at com.google.gwt.dev.shell.ModuleSpacePropertyOracle.getSelectionProperty(ModuleSpacePropertyOracle.java:89)
	at com.google.gwt.dev.cfg.ConditionWhenPropertyIs.doEval(ConditionWhenPropertyIs.java:84)
	at com.google.gwt.dev.cfg.Condition.isTrue(Condition.java:53)
	at com.google.gwt.dev.cfg.ConditionAny.doEval(ConditionAny.java:53)
	at com.google.gwt.dev.cfg.Condition.isTrue(Condition.java:53)
	at com.google.gwt.dev.cfg.ConditionAll.doEval(ConditionAll.java:54)
	at com.google.gwt.dev.cfg.Condition.isTrue(Condition.java:53)
	at com.google.gwt.dev.cfg.Rule.isApplicable(Rule.java:66)
	at com.google.gwt.dev.shell.StandardRebindOracle$Rebinder.getRebindRule(StandardRebindOracle.java:121)
	at com.google.gwt.dev.shell.StandardRebindOracle$Rebinder.rebind(StandardRebindOracle.java:67)
	at com.google.gwt.dev.shell.StandardRebindOracle.rebind(StandardRebindOracle.java:276)
	at com.google.gwt.dev.shell.ShellModuleSpaceHost.rebind(ShellModuleSpaceHost.java:152)
	at com.google.gwt.dev.shell.ModuleSpace.rebind(ModuleSpace.java:613)
	at com.google.gwt.dev.shell.ModuleSpace.rebindAndCreate(ModuleSpace.java:483)
	at com.google.gwt.dev.shell.GWTBridgeImpl.create(GWTBridgeImpl.java:49)
	at com.google.gwt.core.shared.GWT.createImpl(GWT.java:83)
	at com.google.gwt.core.client.GWT.create(GWT.java:86)
	at com.google.gwt.user.client.Window.<clinit>(Window.java:514)
	at com.google.gwt.user.client.Window$Location.getQueryString(Window.java:267)
	at com.google.gwt.user.client.Window$Location.ensureListParameterMap(Window.java:334)
	at com.google.gwt.user.client.Window$Location.getParameter(Window.java:212)
	at com.google.gwt.logging.client.LogConfiguration$LogConfigurationImplRegular.setLevels(LogConfiguration.java:111)
	at com.google.gwt.logging.client.LogConfiguration$LogConfigurationImplRegular.configureClientSideLogging(LogConfiguration.java:72)
	at com.google.gwt.logging.client.LogConfiguration.onModuleLoad(LogConfiguration.java:162)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:601)
	at com.google.gwt.dev.shell.ModuleSpace.onLoad(ModuleSpace.java:423)
	at com.google.gwt.dev.shell.OophmSessionHandler.loadModule(OophmSessionHandler.java:200)
	at com.google.gwt.dev.shell.BrowserChannelServer.processConnection(BrowserChannelServer.java:530)
	at com.google.gwt.dev.shell.BrowserChannelServer.run(BrowserChannelServer.java:368)
	at java.lang.Thread.run(Thread.java:722)
      [ERROR] Unable to get value of property 'user.agent'
com.google.gwt.core.ext.BadPropertyValueException: Missing property 'user.agent' was
not specified
	at com.google.gwt.dev.shell.ModuleSpacePropertyOracle.getConfigurationProperty(ModuleSpacePropertyOracle.java:74)
	at com.google.gwt.dev.cfg.ConditionWhenPropertyIs.doEval(ConditionWhenPropertyIs.java:88)
	at com.google.gwt.dev.cfg.Condition.isTrue(Condition.java:53)
	at com.google.gwt.dev.cfg.ConditionAny.doEval(ConditionAny.java:53)
	at com.google.gwt.dev.cfg.Condition.isTrue(Condition.java:53)
	at com.google.gwt.dev.cfg.ConditionAll.doEval(ConditionAll.java:54)
	at com.google.gwt.dev.cfg.Condition.isTrue(Condition.java:53)
	at com.google.gwt.dev.cfg.Rule.isApplicable(Rule.java:66)
	at com.google.gwt.dev.shell.StandardRebindOracle$Rebinder.getRebindRule(StandardRebindOracle.java:121)
	at com.google.gwt.dev.shell.StandardRebindOracle$Rebinder.rebind(StandardRebindOracle.java:67)
	at com.google.gwt.dev.shell.StandardRebindOracle.rebind(StandardRebindOracle.java:276)
	at com.google.gwt.dev.shell.ShellModuleSpaceHost.rebind(ShellModuleSpaceHost.java:152)
	at com.google.gwt.dev.shell.ModuleSpace.rebind(ModuleSpace.java:613)
	at com.google.gwt.dev.shell.ModuleSpace.rebindAndCreate(ModuleSpace.java:483)
	at com.google.gwt.dev.shell.GWTBridgeImpl.create(GWTBridgeImpl.java:49)
	at com.google.gwt.core.shared.GWT.createImpl(GWT.java:83)
	at com.google.gwt.core.client.GWT.create(GWT.java:86)
	at com.google.gwt.user.client.Window.<clinit>(Window.java:514)
	at com.google.gwt.user.client.Window$Location.getQueryString(Window.java:267)
	at com.google.gwt.user.client.Window$Location.ensureListParameterMap(Window.java:334)
	at com.google.gwt.user.client.Window$Location.getParameter(Window.java:212)
	at com.google.gwt.logging.client.LogConfiguration$LogConfigurationImplRegular.setLevels(LogConfiguration.java:111)
	at com.google.gwt.logging.client.LogConfiguration$LogConfigurationImplRegular.configureClientSideLogging(LogConfiguration.java:72)
	at com.google.gwt.logging.client.LogConfiguration.onModuleLoad(LogConfiguration.java:162)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:601)
	at com.google.gwt.dev.shell.ModuleSpace.onLoad(ModuleSpace.java:423)
	at com.google.gwt.dev.shell.OophmSessionHandler.loadModule(OophmSessionHandler.java:200)
	at com.google.gwt.dev.shell.BrowserChannelServer.processConnection(BrowserChannelServer.java:530)
	at com.google.gwt.dev.shell.BrowserChannelServer.run(BrowserChannelServer.java:368)
	at java.lang.Thread.run(Thread.java:722)
[ERROR] Deferred binding failed for 'com.google.gwt.user.client.impl.WindowImpl'; expect
subsequent failures
[ERROR] Unable to load module entry point class com.google.gwt.logging.client.LogConfiguration
(see associated exception for details)
java.lang.RuntimeException: Deferred binding failed for 'com.google.gwt.user.client.impl.WindowImpl'
(did you forget to inherit a required module?)
	at com.google.gwt.dev.shell.GWTBridgeImpl.create(GWTBridgeImpl.java:53)
	at com.google.gwt.core.shared.GWT.createImpl(GWT.java:83)
	at com.google.gwt.core.client.GWT.create(GWT.java:86)
	at com.google.gwt.user.client.Window.<clinit>(Window.java:514)
	at com.google.gwt.user.client.Window$Location.getQueryString(Window.java:267)
	at com.google.gwt.user.client.Window$Location.ensureListParameterMap(Window.java:334)
	at com.google.gwt.user.client.Window$Location.getParameter(Window.java:212)
	at com.google.gwt.logging.client.LogConfiguration$LogConfigurationImplRegular.setLevels(LogConfiguration.java:111)
	at com.google.gwt.logging.client.LogConfiguration$LogConfigurationImplRegular.configureClientSideLogging(LogConfiguration.java:72)
	at com.google.gwt.logging.client.LogConfiguration.onModuleLoad(LogConfiguration.java:162)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:601)
	at com.google.gwt.dev.shell.ModuleSpace.onLoad(ModuleSpace.java:423)
	at com.google.gwt.dev.shell.OophmSessionHandler.loadModule(OophmSessionHandler.java:200)
	at com.google.gwt.dev.shell.BrowserChannelServer.processConnection(BrowserChannelServer.java:530)
	at com.google.gwt.dev.shell.BrowserChannelServer.run(BrowserChannelServer.java:368)
	at java.lang.Thread.run(Thread.java:722)
Caused by: com.google.gwt.core.ext.UnableToCompleteException: (see previous log entries)
	at com.google.gwt.dev.cfg.ConditionWhenPropertyIs.doEval(ConditionWhenPropertyIs.java:129)
	at com.google.gwt.dev.cfg.Condition.isTrue(Condition.java:53)
	at com.google.gwt.dev.cfg.ConditionAny.doEval(ConditionAny.java:53)
	at com.google.gwt.dev.cfg.Condition.isTrue(Condition.java:53)
	at com.google.gwt.dev.cfg.ConditionAll.doEval(ConditionAll.java:54)
	at com.google.gwt.dev.cfg.Condition.isTrue(Condition.java:53)
	at com.google.gwt.dev.cfg.Rule.isApplicable(Rule.java:66)
	at com.google.gwt.dev.shell.StandardRebindOracle$Rebinder.getRebindRule(StandardRebindOracle.java:121)
	at com.google.gwt.dev.shell.StandardRebindOracle$Rebinder.rebind(StandardRebindOracle.java:67)
	at com.google.gwt.dev.shell.StandardRebindOracle.rebind(StandardRebindOracle.java:276)
	at com.google.gwt.dev.shell.ShellModuleSpaceHost.rebind(ShellModuleSpaceHost.java:152)
	at com.google.gwt.dev.shell.ModuleSpace.rebind(ModuleSpace.java:613)
	at com.google.gwt.dev.shell.ModuleSpace.rebindAndCreate(ModuleSpace.java:483)
	at com.google.gwt.dev.shell.GWTBridgeImpl.create(GWTBridgeImpl.java:49)
	... 18 more
[ERROR] Failed to load module 'com.google.gwt.TestProjectJUnit.JUnit' from user agent
'HtmlUnit-Mozilla/5.0 (Windows NT 6.1; rv:17.0) Gecko/20100101 Firefox/17.0' at localhost:49594
com.google.gwt.core.ext.UnableToCompleteException: (see previous log entries)
	at com.google.gwt.dev.shell.ModuleSpace.onLoad(ModuleSpace.java:461)
	at com.google.gwt.dev.shell.OophmSessionHandler.loadModule(OophmSessionHandler.java:200)
	at com.google.gwt.dev.shell.BrowserChannelServer.processConnection(BrowserChannelServer.java:530)
	at com.google.gwt.dev.shell.BrowserChannelServer.run(BrowserChannelServer.java:368)
	at java.lang.Thread.run(Thread.java:722)
Okt 28, 2014 11:05:54 AM com.gargoylesoftware.htmlunit.WebConsole info
INFO: /launch error: Failed to load module com.google.gwt.TestProjectJUnit.JUnit: undefined
```

Reported by `dankurka@google.com` on 2014-10-28 14:16:42

**Comment by tbroyer** [comment]
Cannot reproduce at HEAD (I've had an OOME, but not that error), I suppose it had been fixed before releasing 2.7.
